helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -103,19 +104,17 @@ void assignNewBatch(Collection<AssignableReplica> replicas) {
     }
 
     // Update the global state after all single replications' calculation is done.
-    for (String key : totalPartitionCapacity.keySet()) {
-      updateCapacityAndUtilization(key, totalPartitionCapacity.get(key));
+    for (String capacityKey : totalPartitionCapacity.keySet()) {
+      updateCapacityAndUtilization(capacityKey, totalPartitionCapacity.get(capacityKey));
     }
   }
 
   /**
    * Assign a replica to the node.
-   * @param assignableReplica - the replica to be assigned
+   * @param replica - the replica to be assigned
    */
-  void assign(AssignableReplica assignableReplica) {
-    addToAssignmentRecord(assignableReplica);
-    assignableReplica.getCapacity().entrySet().stream()
-        .forEach(capacity -> updateCapacityAndUtilization(capacity.getKey(), capacity.getValue()));
+  void assign(AssignableReplica replica) {
+    assign(ImmutableList.of(replica));
```

Comment:
I think the duplicate code is minor. Creating a new list just cause overhead, right?

Reply:
The overhead is minor. Only making one copy of implementation will make the method behavior more stable since any future changes if any, will only need to happen at one single place.